adapter/src/utils/useLocale.js

```diff
@@ -23,22 +52,41 @@ const setGlobalLocale = (locale) => {
     }
     moment.locale(locale)
 
-    const simplifiedLocale = simplifyLocale(locale)
-    i18n.changeLanguage(simplifiedLocale)
+    const resolvedLocale = validateLocaleByBundle(locale)
```

Comment:
maybe a better name for `validateLocaleByBundle` is just `getResolvedLocale` .. that's what it does for me (the validation is an internal detail)